find/main/decorators.py

```diff
@@ -0,0 +1,19 @@
+from functools import wraps
+
+from flask import abort, g
+
+from find.const import InternalDomain
+
+
+def check_internal_user(func):
+    @wraps(func)
+    def decorated_function(*args, **kwargs):
+        internal_domains = tuple(k.value for k in InternalDomain)
+        authenticated = g.is_authenticated
+        is_communities = g.is_authenticated and g.user.email.endswith(internal_domains)
```

Comment:
I think this `.endswith` check is broken - checking if a string ends with a tuple of domains?

Should it be something like this instead?

```suggestion
        is_communities = g.is_authenticated and any(g.user.email.endswith(domain) for domain in internal_domains)
```

Reply:
Huh, TIL that `.endswith` [accepts a tuple of suffixes to look for](https://docs.python.org/3/library/stdtypes.html#str.endswith). That explains why the tests pass.

Carry on!